src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -2476,6 +2478,7 @@ def verify_service_connector_config(
             f"{SERVICE_CONNECTORS}{SERVICE_CONNECTOR_VERIFY}",
             body=service_connector,
             params={"list_resources": list_resources},
+            timeout=timeout,
```

Comment:
This works too, but this isn't really what I meant. I would totally keep implementation details like these away from the user land. You shouldn't even expect the timeout to be provided here as an argument by the external calls. Instead, just enforce a higher value that you already know works (120s is fine IMO):

```suggestion
            timeout=max(self.config.http_timeout, SERVICE_CONNECTOR_VERIFY_REQUEST_TIMEOUT),
```

With `SERVICE_CONNECTOR_VERIFY_REQUEST_TIMEOUT=120` defined in `constants.py` ofc.

The zen store interface, CLI code and any other code for that matter should not be affected or even aware that you're using a higher timeout value here.

I also suggested you use a `max` in case the user does want to configure a custom global timeout value that is higher even than the one you recommend for service connector verification.

Reply:
Reworked